vnext/ReactUWP/Modules/Animated/DecayAnimationDriver.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "DecayAnimationDriver.h"
+#include <jsi/jsi.h>
+#include <math.h>
+
+namespace react { namespace uwp {
+  DecayAnimationDriver::DecayAnimationDriver(
+    int64_t id,
+    const std::shared_ptr<ValueAnimatedNode>& animatedValue,
+    const Callback& endCallback,
+    const folly::dynamic& config)
+    : AnimationDriver(id, animatedValue, endCallback, config)
+  {
+    m_deceleration = config.find(s_decelerationName).dereference().second.asDouble();
+    m_velocity = config.find(s_velocityName).dereference().second.asDouble();
+  }
+
+  std::tuple<winrt::CompositionAnimation, winrt::CompositionScopedBatch> DecayAnimationDriver::MakeAnimation(const folly::dynamic& config)
+  {
+    const auto [scopedBatch, animation] = []()
+    {
+      const auto compositor = winrt::Window::Current().Compositor();
+      return std::make_tuple(compositor.CreateScopedBatch(winrt::CompositionBatchTypes::AllAnimations), compositor.CreateScalarKeyFrameAnimation());
+    }();
+
+    std::chrono::milliseconds duration(static_cast<int>(m_velocity / -m_deceleration * 1000));
```

Comment:
should there be some > 0 check on m_deceleration somewhere in this func or ctor, to not divide by 0?